src/app/users.service.ts

```diff
@@ -1,33 +1,40 @@
 import { Injectable } from '@angular/core';
-import { usersMock } from './users.mock';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/operators';
 import { UserStructure } from './user-structure.model';
+import { Observable } from 'rxjs';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root'                                                                    
 })
 export class UsersService {  
 
-  users: UserStructure[] = usersMock;
+  constructor(private http: HttpClient){}
 
-  getUsers(): UserStructure[]{
-    return this.users;
+  getUsers(): Observable<UserStructure[]>{
```

Comment:
`UserStructure` can be renamed to `User`. Use the same model that you have created earlier